arduino-ide-extension/src/node/i18n/localization-backend-contribution.ts

```diff
@@ -0,0 +1,39 @@
+import * as express from 'express';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { LocalizationBackendContribution as TheiaLocalizationBackendContribution } from '@theia/core/lib/node/i18n/localization-backend-contribution';
+import { PluginDeployer } from '@theia/plugin-ext/lib/common/plugin-protocol';
+import { PluginDeployerImpl } from '@theia/plugin-ext/lib/main/node/plugin-deployer-impl';
+import { Deferred } from '@theia/core/lib/common/promise-util';
+
+@injectable()
+export class LocalizationBackendContribution extends TheiaLocalizationBackendContribution {
+  @inject(PluginDeployer)
+  protected readonly pluginDeployer: PluginDeployerImpl;
+
+  private initialized = new Deferred<void>();
+
+  override async initialize(): Promise<void> {
+    this.pluginDeployer.onDidDeploy(() => {
+      this.initialized.resolve();
+    });
+    await this.localizationRegistry.initialize();
+  }
+
+  override configure(app: express.Application): void {
+    app.get('/i18n/:locale', async (req, res) => {
+      let locale = req.params.locale;
+      /*
+        Waiting for the deploy of the language plugins is neecessary to avoid checking the available
+        languages before they're finished to be loaded: https://github.com/eclipse-theia/theia/issues/11471 
+      */
+      await this.initialized.promise;
```

Comment:
Just wondering, did you check if it slows down the IDE2 startup? The frontend must wait for this, right?

Reply:
I'm not sure how to calculate this.

With the current build, I can see these two log lines:
```
root INFO [2c0ce3a6-ca5a-4f40-b831-533e06823a9f] Load contributions of 16 plugins: 86.5 ms [Finished 4.873 s after frontend start]
root INFO [2c0ce3a6-ca5a-4f40-b831-533e06823a9f] Start of 16 plugins: 575.4 ms [Finished 5.460 s after frontend start]
```

Then, in this case, I think the frontend is slowed down by `86.5 ms + 574.4 ms = 660.9 ms`

Am I correct?

Reply:
```ts
const start = performance.now();
await this.initialized.promise;
console.log('took: ' + (performance.now() - start) + ' ms')
```

Reply:
```
root INFO took: 405.16497199982405 ms
```

Quite significant. I suppose this delay will increase as we keep adding new language packs.

Reply:
> This should be done fairly quickly without impacting the load time on Electron
from: [here](https://github.com/eclipse-theia/theia/pull/11472#pullrequestreview-1053143796)

Reply:
maybe it's worth measuring the impact in the build (if the above is from source)? (maybe there are some optimisations that might speed up the plugin deployment)

Reply:
I've pushed the code to log the performance so that we see how the time grows as we keep adding new language packs.